### PR TITLE
Feature/improve sidebar

### PR DESCRIPTION
**Pull request by thaihoa-fe**

### CODE REVIEW

File: src/components/TableOfContents/index.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    top: auto;
+    border: none;
+    margin: 0;
+    position: relative;
+  }
+`;
+
+function TableOfContents({ content }) {
+  const [showTOC, setShowTOC] = useState(false);
+  let mounted = false;
```
Comment (nhahoangts) [inline src/components/TableOfContents/index.js]: What is the purpose of this flag? I didn't see it is used anywhere else?

File: src/components/TableOfContents/index.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  const handleToggleTOC = () => {
+    setShowTOC(!showTOC);
+  };
+  useEffect(() => {
+    document.documentElement.addEventListener('click', closePopup);
+    return () => {
+      document.documentElement.removeEventListener('click', closePopup);
+      mounted = false;
+    };
+  }, mounted);
```
Comment (nhahoangts) [inline src/components/TableOfContents/index.js]: Should the dependencies list is array?

File: src/components/TableOfContents/index.js (hunk 1 above)
Comment (thaihoa-fe) [reply src/components/TableOfContents/index.js]: fixed

File: src/components/TableOfContents/index.js (hunk 2 above)
Comment (thaihoa-fe) [reply src/components/TableOfContents/index.js]: fixed

**Pull request merged**